docs/permission/writing-a-policy.md

```diff
@@ -38,36 +38,50 @@ Let's change the policy to the following:
 
 ```diff
 - import { IdentityClient } from '@backstage/plugin-auth-node';
-+ import { BackstageIdentityResponse, IdentityClient } from '@backstage/plugin-auth-node';
-+ import { catalogConditions, createCatalogPolicyDecision } from '@backstage/plugin-catalog-backend';
++ import {
++   BackstageIdentityResponse,
++   IdentityClient
++ } from '@backstage/plugin-auth-node';
+  import {
+  AuthorizeResult,
+  PolicyDecision,
++ isResourcePermission,
+} from '@backstage/plugin-permission-common';
++ import {
++   catalogConditions,
++   createCatalogConditionalDecision,
++ } from '@backstage/plugin-catalog-backend';
 
   ...
 
   class TestPermissionPolicy implements PermissionPolicy {
--   async handle(request: PolicyAuthorizeQuery): Promise<PolicyDecision> {
+-   async handle(request: PolicyQuery): Promise<PolicyDecision> {
 +   async handle(
-+     request: PolicyAuthorizeQuery,
++     request: PolicyQuery,
 +     user?: BackstageIdentityResponse,
 +    ): Promise<PolicyDecision> {
-      if (request.permission.name === 'catalog.entity.delete') {
--       return {
--         result: AuthorizeResult.DENY,
--       };
-+       return createCatalogPolicyDecision(
-+         catalogConditions.isEntityOwner(
-+           user?.identity.ownershipEntityRefs ?? [],
-+         ),
-+       );
-      }
++     if (isResourcePermission(request.permission, 'catalog-entity')) {
+        if (request.permission.name === 'catalog.entity.delete') {
```

Comment:
```suggestion
 -    if (request.permission.name === 'catalog.entity.delete') {
```

Reply:
Thoughts on this:
1. Demonstrate `isPermission(...)` here with a single `if`
2. Add another `isPermission(...)` to check for `catalog.entity.read`
3. Introduce `isResource` as a way to provide one conditional response for all catalog resource permissions

Reply:
@Joonpark13 I just pushed fde67142d4db1d23d603e89056800095ed71852d to address some of our remaining comments. Lmk what you think!

Reply:
I like the direction you've taken it! ✅